### PR TITLE
Alternatives output into parser module 1.x

### PR DESCRIPTION
**Pull request by PaulWay**
There's a lot of information possible in the output of `alternatives --display`, and this parser class captures it all.  This also allows other parsers, say for `alternatives --display mta`.  It allows for more analysis of the output to find problems.

This also deprecates the `default_java` parser function, whose output is now available in the `link` property of the JavaAlternatives class.

### CODE REVIEW

File: insights/parsers/tests/test_alternatives.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.111-1.b15.el7_2.x86_64/jre/bin/java - priority 1800111
+/usr/lib/jvm/jre-1.6.0-ibm.x86_64/bin/java - priority 16091
+ slave ControlPanel: /usr/lib/jvm/jre-1.6.0-ibm.x86_64/bin/ControlPanel
+ slave keytool: /usr/lib/jvm/jre-1.6.0-ibm.x86_64/bin/keytool
+ slave policytool: /usr/lib/jvm/jre-1.6.0-ibm.x86_64/bin/policytool
+ slave rmid: /usr/lib/jvm/jre-1.6.0-ibm.x86_64/bin/rmid
+Current `best' version is /usr/lib/jvm/jre-1.6.0-ibm.x86_64/bin/java.
+"""
+
+alter_java = """
```
Comment (xiangce) [inline insights/parsers/tests/test_alternatives.py]: This variable is not used, better remove it.
All others look good to me.

File: insights/parsers/alternatives.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+            elif line.startswith(' slave ') and len(words) == 3 and current_path:
+                # line: ' slave ControlPanel: /usr/lib/jvm/jre-1.6.0-ibm.x86_64/bin/ControlPanel'
+                current_path['slave'][words[1][:-1]] = words[2]  # remove final : from program
+            elif line.startswith("Current `best' version is ") and len(words) == 5:
+                # line: 'Current `best' version is /usr/lib/jvm/jre-1.6.0-ibm.x86_64/bin/java.'
+                self.best = words[4][:-1]  # strip trailing . from path
+
+        # Store program's current path if both are available
+        if self.program and current_path:
+                self.paths.append(current_path)
```
Comment (bfahr) [inline insights/parsers/alternatives.py]: Indentation is incorrect here.

File: insights/parsers/tests/test_alternatives.py (hunk 1 above)
Comment (PaulWay) [inline insights/parsers/tests/test_alternatives.py]: Thanks @xiangce - will do.

File: insights/parsers/alternatives.py (hunk 2 above)
Comment (PaulWay) [inline insights/parsers/alternatives.py]: I'm not sure whether you think they should be indented or outdented, but this needs to occur even if the output does not contain the current 'best' version.

However, I've realised that I can simplify this parsing a bit - let me know what you think of it now.

**Comment by xiangce** [comment]
[Alternatives output into parser module 1.x](https://trello.com/c/P3zJCIeI/1-alternatives-output-into-parser-module-1x)

**Pull request merged**